src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -80,15 +82,15 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
     errorSubscription: Subscription;
 
     isProgrammingExercise() {
-        return this.activeExercise.type === ExerciseType.PROGRAMMING;
+        return !this.activeExamPage.isOverviewPage && this.activeExamPage.exercise!.type === ExerciseType.PROGRAMMING;
     }
 
     isProgrammingExerciseWithCodeEditor(): boolean {
-        return this.isProgrammingExercise() && (this.activeExercise as ProgrammingExercise).allowOnlineEditor === true;
+        return this.isProgrammingExercise() && !this.activeExamPage.isOverviewPage && (this.activeExamPage.exercise as ProgrammingExercise).allowOnlineEditor === true;
     }
 
     isProgrammingExerciseWithOfflineIDE(): boolean {
-        return this.isProgrammingExercise() && (this.activeExercise as ProgrammingExercise).allowOfflineIde === true;
+        return this.isProgrammingExercise() && !this.activeExamPage.isOverviewPage && (this.activeExamPage.exercise as ProgrammingExercise).allowOfflineIde === true;
```

Comment:
isProgramminExercise already checks for !this.activeExamPage.isOverviewPage so no need for it here.

Reply:
done